ctt/process_dependencies.py

```diff
@@ -523,6 +472,146 @@ def create_backwards_compatible_cfg(
+    }
+    shared_uploaders = {
+        name: _uploader(cfg) for name, cfg in processing_cfg.get('uploaders', {}).items()
+    }
+
+    def create_replication_plan_resource(
+        component: ocm.Component,
+        resource: ocm.Resource,
+    ) -> ctt.model.ReplicationPlanResource | None:
+        for pipeline in enum_processing_cfgs(
+            processing_cfg=processing_cfg,
+            shared_targets=shared_targets,
+            shared_processors=shared_processors,
+            shared_uploaders=shared_uploaders,
+        ):
+            replication_plan_resource = pipeline.process(
+                component=component,
+                resource=resource,
+                tgt_oci_registry=tgt_oci_registry,
+                oci_client=oci_client,
+                replication_mode=replication_mode,
+            )
+
+            if replication_plan_resource:
+                return replication_plan_resource
+
+        logger.debug(
+            f'skipped processing: {component.name}:{resource.access} ({tgt_oci_registry=})'
+        )
+
+    components_with_resource = [
+        (component_descriptor.component, resource)
+        for component_descriptor in component_descriptors
+        for resource in component_descriptor.component.resources
+        if resource.access.type is ocm.AccessType.OCI_REGISTRY
+    ]
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=16)
+    yield from (
+        replication_plan_resource for replication_plan_resource in executor.map(
+            create_replication_plan_resource,
+            [component for component, _ in components_with_resource],
+            [resource for _, resource in components_with_resource],
+        ) if replication_plan_resource
+    )
+
+
+def create_replication_plan_step(
+    processing_cfg: dict,
+    root_component_descriptor: ocm.ComponentDescriptor,
+    component_descriptor_lookup: cnudie.retrieve.ComponentDescriptorLookupById,
+    tgt_oci_registry: str,
+    tgt_ocm_repository: str,
+    oci_client: oci.client.Client,
+    delivery_service_client: delivery.client.DeliveryServiceClient,
+    replication_mode: oci.ReplicationMode=oci.ReplicationMode.PREFER_MULTIARCH,
+    component_filter: collections.abc.Callable[[ocm.Component], bool] | None=None,
+    reftype_filter: collections.abc.Callable[[cnudie.iter.NodeReferenceType], bool] | None=None,
+    remove_label: collections.abc.Callable[[str], bool]=None,
+) -> ctt.model.ReplicationPlanStep:
+    tgt_ocm_repo = ocm.OciOcmRepository(
+        baseUrl=tgt_oci_registry,
+        subPath=tgt_ocm_repository,
+    )
+
+    tgt_component_descriptor_lookup = cnudie.retrieve.create_default_component_descriptor_lookup(
+        ocm_repository_lookup=cnudie.retrieve.ocm_repository_lookup(tgt_ocm_repo.oci_ref),
+        oci_client=oci_client,
+        delivery_client=delivery_service_client,
+        fallback_to_service_mapping=False,
+    )
+
+    component_descriptors = tuple(determine_changed_components(
```

Comment:
we might - if this is more convenient - keep references to `ocm.Component` (rather than ComponentDescriptor), and create the `ComponentDescriptor`- envelope ad-hoc during upload.